pxtlib/util.ts

```diff
@@ -1429,6 +1429,8 @@ namespace ts.pxtc.BrowserImpl {
 
             let headers = Util.clone(options.headers) || {}
 
+            options.url = /^http/.test(options.url) ? options.url : pxt.Cloud.appRoot + options.url;
```

Comment:
When is this needed?

Reply:
For routing to the configured backend when the url is relative (see my changes in cloud.ts -- the `BACKEND` field). I needed this so that calls to endpoints like `/auth/login` are able to hit the locally running backend.